src/app/components/AnalyticsConsent/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { useEffect, useState } from 'react'
+import { useLocation } from 'react-router-dom'
+import Snackbar from '@mui/material/Snackbar'
+import Typography from '@mui/material/Typography'
+import { styled } from '@mui/material/styles'
+import Button from '@mui/material/Button'
+import Link from '@mui/material/Link'
+import { Trans, useTranslation } from 'react-i18next'
+import Card from '@mui/material/Card'
+import CardContent from '@mui/material/CardContent'
+import CardActions from '@mui/material/CardActions'
+import { useScreenSize } from 'app/hooks/useScreensize'
+import * as matomo from './initializeMatomo'
+import { legalDocuments } from '../../utils/externalLinks'
+
+matomo.loadMatomo()
+
+const AcceptCookiesButton = styled(Button)(({ theme }) => ({
+  paddingLeft: theme.spacing(5),
+  paddingRight: theme.spacing(5),
+}))
+
+const DeclineCookiesButton = styled(Button)(({ theme }) => ({
+  paddingLeft: theme.spacing(5),
+  paddingRight: theme.spacing(5),
+}))
+
+export const AnalyticsConsent = () => {
+  const { t } = useTranslation()
+  const { isMobile } = useScreenSize()
+  const [hasAccepted, setHasAccepted] = useState<matomo.HasAccepted>(matomo.hasAccepted())
+
+  useEffect(() => {
+    window._paq.push([
+      function () {
+        // Matomo loaded
+        setHasAccepted(matomo.hasAccepted())
+        // TODO: bug: still sometimes gets stuck with 'error_matomo_not_loaded' ???
+        console.log('loaded', matomo.hasAccepted())
+      },
+    ])
+  }, [])
+
+  const location = useLocation()
+  const [previousURL, setPreviousURL] = useState(document.referrer)
+  useEffect(() => {
+    // Warning: When not opted in, window._paq array will grow indefinitely.
```

Comment:
Can we check here if user has opted in? Otherwise just ignore the code bellow. Don't see I reason, why we would track if user is not opted in. Unless the reasoning is that after you opt-in we will gathered all the data we collected, after user opts in. But in that case I don't like the solution - not even sure what the legal position is in that case.

Reply:
+1

Looks like the code comment doesn't apply when using matomo's consent storage (comment was copied from localStorage PR). Matomo automatically throws away events that were pushed before consent. But I'll add an explicit `if consented`.

(Theoretically user should be able to consent to sending navigations that happened before consent, but I don't think we inform them sufficiently.)